tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -231,7 +231,11 @@ static PyObject * THPVariable_numel(PyObject* self, PyObject* args)
      return handle_torch_function(self, "numel", args);
    }
    auto& self_ = THPVariable_Unpack(self);
-   return THPUtils_packInt64(self_.numel());
+   if (jit::tracer::isTracing()) {
```

Comment:
Why does this needs to be special case. This is not something we usually do for any other op right?

Reply:
After reading similar snippet for `size()` (and https://github.com/pytorch/pytorch/pull/6554#issuecomment-380927254), my understanding is that operators that produce primitive typed outputs cannot be traced. Hence the special case of changing to return pytorch scalar instead of python scalar under tracing mode.